app/src/utils/routingUtils.ts

```diff
@@ -1,19 +1,40 @@
 export const RETURN_URL_URL_PARAM = "returnUrl";
 
 /**
- * Gets the current url path and query string, excluding the domain and protocol.
+ * Creates a URL to redirect to the login page with the current path as the return URL
  */
-const getCurrentUrlPath = (): string => {
-  return window.location.pathname + window.location.search;
-};
+export function createLoginRedirectUrl() {
+  const path = window.location.pathname;
+  const basename = window.Config.basename;
+  const searchParams = window.location.search;
+  let basePath = "";
+  let pathAfterBase = "";
+  if (basename === "" || basename === "/") {
+    pathAfterBase = path;
+  } else if (basename.startsWith("/") && path.startsWith(basename)) {
+    const afterBase = path.slice(basename.length);
+    if (afterBase === "" || afterBase.startsWith("/")) {
+      basePath = basename;
+      pathAfterBase = afterBase;
+    }
```

Comment:
I'm a little confused here:
The check on line 14:
Is this to make sure the basename is valid? Are you protecting against someone setting their basename as "noSlashUrlBase"?

On the check on line 16.
1. What's the purpose of this check? To block against invalid after bases? It feels like we would want to always just append the afterbase?
2. It seems maybe like it won't catch all cases:
Consider this case
```javascript
const basename = "/test"
const path = "/test?selectedSpanId=123" // valid

const afterBase = path.slice(basename.length);
if (afterBase === "" || afterBase.startsWith("/")) {
   console.log("this is never hit")
}
```

Naively it seems like the logic here would just be, but that might not be sufficient:
```javascript
const pathWithoutBaseName =
  basename != null && path.startsWith(basename)
    ? path.slice(basename.length)
    : path
```

Reply:
> `const path = "/test?selectedSpanId=123"`

`window.location.pathname` does not include query parameters

Reply:
My goal with the check is to make sure whatever winds up inside `returnUrl` starts with `/`. You could imagine someone having base name configured as `/phoenix` and they navigate to `/phoenix_path` in the UI. Then the relative URL would be `_path` if we did not perform the check.